src/AppBundle/Security/PasswordListener.php

```diff
@@ -27,15 +28,15 @@ public function prePersist(LifecycleEventArgs $args)
     {
         $entity = $args->getEntity();
 
-        if ($entity instanceof User && $entity->getPlainPassword()) {
+        if ($entity instanceof User || $entity instanceof Admin && $entity->getPlainPassword()) {
```

Comment:
Admin is not a User?
Simplify this and below condition.
Don't use complex condition - use few simple and return; statement, e.g.
```php
if (false === in_array(UserInterface::class, class_implements($entity)) {
    return;
}

$this->encodePassword($entity);
```